account_payment_term_partner_holiday/models/account_payment_term.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from dateutil.relativedelta import relativedelta
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    @api.one
+    def compute(self, value, date_ref=False):
+        result = super().compute(value=value, date_ref=date_ref)[0]
+        if self.env.context.get("default_partner_id"):
```

Comment:
Is this context the proper one? I think you should inject an specific context from invoices when doing this computation, as I think it's not assured to have this context always (think also on other automations like sales invoicing).

Reply:
You are right, the name of the context is not the most appropriate but it is something that is launched automatically (https://github.com/odoo/odoo/blob/3353fef07a3a54596b30aabc29d9729f2eb3875e/addons/account/models/account_invoice.py#L2057). You mean `_onchange_payment_term_id()` should be overwritten and add custom context to use in `compute()` function?
I think it is important to note that this function and `_onchange_date_due()` do "the same".

Reply:
I mean that you inject a specific context key in the methods that call this one:

https://github.com/odoo/odoo/blob/a2463b213e7f19b4a089707b226f9a52b4a52b7a/addons/account/models/account_invoice.py#L908
https://github.com/odoo/odoo/blob/a2463b213e7f19b4a089707b226f9a52b4a52b7a/addons/account/models/account_invoice.py#L1301

so you assure that you don't depend on external agents.

Reply:
This is still pending.